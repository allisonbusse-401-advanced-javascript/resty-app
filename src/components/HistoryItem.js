import React from 'react';
// import styles from './Header.css';
import PropTypes from 'prop-types';

const HistoryItem = ({ url, method, storeKey, onHistoryClick }) => (

  <button onClick={onHistoryClick} storekey={storeKey}>
    <h2>{method}</h2>
    <p>{url}</p>
  </button>

);

HistoryItem.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  storeKey: PropTypes.string.isRequired,
  onHistoryClick: PropTypes.func.isRequired
};


export default HistoryItem;
