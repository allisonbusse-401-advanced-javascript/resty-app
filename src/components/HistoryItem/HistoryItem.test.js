import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  const props = {
    handleHistoryClick: () => {},
    url: 'jkl',
    method: 'jkl',
    headers: {},
    results: {}
  };
  it('renders HistoryItem', () => {   
    const wrapper = shallow(<HistoryItem {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
