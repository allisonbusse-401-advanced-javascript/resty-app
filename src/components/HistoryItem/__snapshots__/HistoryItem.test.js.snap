// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<div
  onClick={[Function]}
>
  <h2>
    jkl
  </h2>
  <p>
    jkl
  </p>
</div>
`;
